### PR TITLE
[WFLY-15767] Update OpenTelemetry to 1.9.1

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-15767

### CODE REVIEW

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/trace/main/module.xml [hunk 1]
```diff
@@ -32,7 +32,6 @@
         <artifact name="${io.opentelemetry:opentelemetry-exporter-jaeger}"/>
         <artifact name="${io.opentelemetry:opentelemetry-exporter-otlp-common}"/>
         <artifact name="${io.opentelemetry:opentelemetry-exporter-otlp-trace}"/>
-        <artifact name="${io.opentelemetry:opentelemetry-proto}"/>
```
Comment (bstansberry) [inline ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/trace/main/module.xml:35]: Are we going to replace this with something from https://github.com/open-telemetry/opentelemetry-proto-java? The OTEL v 1.7 release notes say something about that.

Comment (jasondlee) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/trace/main/module.xml:35]: It does, but we don't need it (no idea why it was added, tbh). I started the server and ran manual tests with my otel demo app. Traces were recorded and exported to Jaeger as expected, and all tests in basic/ and manualmode/ are passing with no errors/complaints in the logs, so I don't think we need it all.

Comment (bstansberry) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/trace/main/module.xml:35]: @jasondlee Can users somehow trigger use of this 'Jaeger gRPC exporter' thing that this lib seems to back, i.e. via the APIs we expose? Or is that something our subsystem would have to do for them?

Comment (jasondlee) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/trace/main/module.xml:35]: The export of traces is done automatically by otel itself, based on some config params. It pushes the data to the external collecter (Jaeger, Otel Collector, etc) without any interaction from the user. Is that what you're asking?

Comment (bstansberry) [reply ee-feature-pack/common/src/main/resources/modules/system/layers/base/io/opentelemetry/trace/main/module.xml:35]: @jasondlee coming back to this; sorry for the slow reply

What I was trying to get at was a distinction between what we test vs what a user can do (without obvious hacking). If users could do things without obvious hacking, then we have to think harder about taking those things away. But if there's no API we expose to application code that lets that code control this, and our subsystem has a clear configuration model and that provides no way to control this, then there's no concern.

**Pull request merged**